### ISSUE TITLE
str does not parenthesize subexpressions

### ISSUE THREAD

**Issue by RhysU** [cvxgrp/cvxpy#336]
I expect `10 * (var0 + 5)` below but see something different:
```
In [1]: import cvxpy

In [2]: a = cvxpy.Variable()

In [3]: b = a + 5

In [4]: c = 10 * b

In [5]: c
Out[5]: Expression(AFFINE, UNKNOWN, (1, 1))

In [6]: str(c)  # Seemingly wrong
Out[6]: '10 * var0 + 5'

In [7]: a.value = 1

In [8]: c.value  # Value as expected
Out[8]: 60

In [9]: cvxpy.__version__
Out[9]: '0.4.8'
```

**Comment by SteveDiamond** [comment]
Yes, this has been a known bug for quite a while. I don't find the ``str(expression)`` outputs useful, so I haven't bothered to fix it. You could fix it with a PR by checking if a subexpression should be parenthesized in the ``__str__`` method for the arithmetic operators in https://github.com/cvxgrp/cvxpy/tree/master/cvxpy/atoms/affine

**Comment by rileyjmurray** [comment]
This is resolved in #565.
